### ISSUE TITLE
Support apilinks links with default label text

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6936]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6936|
|Type|defect|
|Created|2014-01-30 00:41:18Z|
|Branch|https://github.com/twisted/twisted/tree/sphinx-api-role-fix-6936|

```
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 195, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/lib/pymodules/python2.7/docutils/statemachine.py", line 237, in run
    context, state, transitions)
  File "/usr/lib/pymodules/python2.7/docutils/statemachine.py", line 458, in check_line
    return method(match, context, next_state)
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 2723, in text
    paragraph, literalnext = self.paragraph(lines, startline)
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 421, in paragraph
    textnodes, messages = self.inline_text(text, lineno)
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 430, in inline_text
    return self.inliner.parse(text, lineno, self.memo, self.parent)
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 516, in parse
    lineno)
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 775, in interpreted_or_phrase_ref
    lineno)
  File "/usr/lib/pymodules/python2.7/docutils/parsers/rst/states.py", line 831, in interpreted
    nodes, messages2 = role_fn(role, rawsource, text, lineno, self)
  File "/home/exarkun/Projects/Twisted/branches/x-forwarded-for-logging-1468/docs/_extensions/apilinks.py", line 36, in make_api_link
    node = nodes.reference(rawtext, utils.unescape(label), refuri=ref,
UnboundLocalError: local variable 'label' referenced before assignment
The full traceback has been saved in /tmp/sphinx-err-1tmPoO.log, if you want to report the issue to the developers.
```

Presumably this is due to some markup error but the exception doesn't actually say what the error is so ... I don't know what the error is.

Attachments:

* [sphinx-err-1tmPoO.log](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/78c/78cf917a201bf207b1136fc9cfae0e3897c386fa/3e87b836325529c1b0ff8c054e438331b542d315.log) (3904 bytes) - added by exarkun on 2014-01-30 00:41:37Z - sphinx error log
* [access-logging.rst](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/78c/78cf917a201bf207b1136fc9cfae0e3897c386fa/21a9ccfceb0d67e71e5313606ef5fa8e54dd2a9d.rst) (3107 bytes) - added by exarkun on 2014-01-30 00:42:29Z - the document I was working on, presumably containing the error

<details><summary>Searchable metadata</summary>

```
trac-id__6936 6936
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_sphinx_api_role_fix_6936 branches/sphinx-api-role-fix-6936
branch_author__julian julian
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1391042478000000 1391042478000000
changetime__1391045395000000 1391045395000000
version__None None
owner__Julian Julian

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph set owner to @khorn|
|-|-|

Looks like there's definitely a branch in apilinks.py where the local `label` variable is not set.

Kevin, can you take a look at this please?

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)|@Julian set owner to @Julian|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|julian commented|
|-|-|

(In [41484]) Branching to sphinx-api-role-fix-6936.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_4"></a>|@Julian removed owner|
|-|-|

Here's the small fix.

Uses the reference as the label if no label is provided explicitly.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph set owner to @Julian|
|-|-|

Needs a NEWS file, but should definitely be '.misc'.

I'm not going to worry about the fact that this doesn't add a test, because we do not distribute this code and really should not be maintaining it.  One of the benefits of sphinx was supposed to be that we wouldn't have to maintain a documentation system, and this feels a whole lot like maintaining a documentation system ;-).  I hope we can pawn these extensions off on someone else and not maintain them ourselves for much longer.

For now: please land.

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|julian set status to `closed`|
|-|-|

(In [41489]) Merge sphinx-api-role-fix-6936: Fix apilinks without labels.

Author: Julian
Reviewers: glyph
Fixes: [#6936](https://github.com/twisted/twisted/issues/6936)
